tick/inference/hawkes_sumgaussians.py

```diff
@@ -0,0 +1,404 @@
+        stopping criterion is below it). If not reached it does ``max_iter``
+        iterations
+
+    verbose : `bool`, default=False
+        If `True`, we verbose things
+
+    n_threads : `int`, default=1
+        Number of threads used for parallel computation.
+
+        * if `int <= 0`: the number of physical cores available on the CPU
+        * otherwise the desired number of threads
+
+    print_every : `int`, default=10
+        Print history information when ``n_iter`` (iteration number) is
+        a multiple of ``print_every``
+
+    record_every : `int`, default=10
+        Record history information when ``n_iter`` (iteration number) is
+        a multiple of ``record_every``
+
+    Other Parameters
+    ----------------
+    approx : `int`, default=0 (read-only)
+        Level of approximation used for computing exponential functions
+
+        * if 0: no approximation
+        * if 1: a fast approximated exponential function is used
+
+    em_max_iter : `int`, default=30
+        Maximum number of loop for inner em algorithm.
+
+    em_tol : `float`, default=None
+        Tolerance of loop for inner em algorithm. If relative difference of
+        baseline and adjacency goes bellow this tolerance, em inner loop
+        will stop.
+        If None, it will be set given a heuristic which look at last
+
+    Attributes
+    ----------
+    n_nodes : `int`
+        Number of nodes / components in the Hawkes model
+
+    n_gaussians : `int`
+        Number of gaussians used to approximate each kernel
+
+    baseline : `np.array`, shape=(n_nodes,)
+        Inferred baseline of each component's intensity
+
+    amplitudes : `np.ndarray`, shape=(n_nodes, n_nodes, n_gaussians)
+        Inferred adjacency matrix
+
+    References
+    ----------
+    Xu, Farajtabar, and Zha (2016, June) in ICML
+    `Learning Granger Causality for Hawkes Processes`
+    <http://jmlr.org/proceedings/papers/v48/xuc16.pdf>`_.
+    """
+
+    _attrinfos = {
+        "_learner": {"writable": False},
+        "_model": {"writable": False},
+        "n_gaussians": {
+            "cpp_setter": "set_n_gaussians"
+        },
+        "em_max_iter": {
+            "cpp_setter": "set_em_max_iters"
+        },
+        "max_mean_gaussian": {
+            "cpp_setter": "set_max_mean_gaussian"
+        },
+        "step_size": {
+            "cpp_setter": "set_step_size"
+        },
+        "penalty_param_lasso": {
+            "cpp_setter": "set_penalty_param_lasso"
+        },
+        "penalty_param_grouplasso": {
+            "cpp_setter": "set_penalty_param_grouplasso"
+        },
+        "_C": {"writable": False},
+        "baseline": {"writable": False},
+        "amplitudes": {"writable": False},
+        "_prox_l1": {"writable": False},
+        "_lasso_grouplasso_ratio": {"writable": False},
+        "approx": {"writable": False}
+    }
+
+    def __init__(self, n_gaussians, max_mean_gaussian=5, step_size=1e-7,
```

Comment:
I think we can have a default value to `n_gaussians` (it is almost like a kernel_size in EM) but I find that `max_mean_gaussian`'s default value (5) is very arbitrary and hence we should force the user to set it manually.